packages/solid-plotly.js/src/factory.tsx

```diff
@@ -55,221 +55,98 @@ const updateEvents: PlotlyHTMLElementEventName[] = [
-      const el = plotEl()
-      handlers[eventName] = handler
-      if (isBasicEvent(eventName)) {
-        el?.on(eventName, () => handler(getCurrentFigure(), el))
       }
-    }
 
-    const removeEventHandler = (eventName: PlotlyHTMLElementEventName) => {
-      const el = plotEl()
-      const handler = handlers[eventName]
-      if (handler && isBasicEvent(eventName)) {
-        el?.removeListener(eventName, () => handler(getCurrentFigure(), el))
-        delete handlers[eventName]
-      }
-    }
-
-    // Type guard to check if an event is a basic event
-    const isBasicEvent = (eventName: PlotlyHTMLElementEventName): eventName is PlotlyBasicEvent => {
-      return updateEvents.includes(eventName)
-    }
-
-    const getCurrentFigure = (): PlotlyFigure => {
-      const el = plotEl()
-      return {
-        data: el?.data as Partial<PlotData>[],
-        layout: el?.layout ?? {},
-        frames: el?._transitionData?._frames,
-      }
-    }
-
-    const updatePlotly = async (
-      shouldInvokeResizeHandler: boolean,
-      callback?: EventHandler,
-      shouldAttachUpdateEvents: boolean = false,
-    ) => {
-      try {
-        const el = plotEl()
-        if (!el) {
-          throw new Error('Missing element reference')
-        }
-
-        await Plotly.react(el, props.data, props.layout || {}, props.config)
-
-        syncWindowResize(shouldInvokeResizeHandler)
-        syncEventHandlers()
-        figureCallback(callback)
-
-        if (shouldAttachUpdateEvents) {
-          attachUpdateEvents()
-        }
-      } catch (err) {
-        if (props.onError && err instanceof Error) {
-          props.onError(err)
-        } else {
-          // eslint-disable-next-line no-console
-          console.error('PlotlyComponent error:', err)
+      const updatePlotly = async (callback?: EventHandler) => {
+        try {
+          await Plotly.react(el, props.data, props.layout || {}, props.config)
+          callback?.(getCurrentFigure(), el)
+        } catch (err) {
+          if (props.onError && err instanceof Error) {
+            props.onError(err)
+          } else {
+            // eslint-disable-next-line no-console
+            console.error('PlotlyComponent error:', err)
+          }
         }
       }
-    }
 
-    const retryUpdatePlotly = () => {
-      const el = plotEl()
-      if (el) {
-        updatePlotly(true, props.onInitialized, true)
-      } else {
-        setTimeout(retryUpdatePlotly, 0) // Retry after 1 tick
-      }
+      onMount(() => {
+        initEventHandlers()
+        initUpdateEvents()
+        updatePlotly(props.onInitialized)
+        createEffect(() => updatePlotly(props.onUpdate), { defer: true })
+
+        createEffect(() => {
+          if (isBrowser && props.useResizeHandler) {
+            const resizeHandler = () => Plotly.Plots.resize(el)
+            window.addEventListener('resize', resizeHandler)
+            onCleanup(() => window.removeEventListener('resize', resizeHandler))
```

Comment:
Is this the same `onCleanup` for the component life cycle? I didn't realize you could define this multiple times in different scopes/closures🤯

Reply:
yes! well, technically there is no component lifecycle. what we think of a component unmounting is actually an effect cleaning up :-) rendering in solid is literally a side-effect